anthos-attached-clusters/aks/main.tf

```diff
@@ -20,9 +20,6 @@ locals {
   }
 }
 
-data "azurerm_subscription" "current" {
-}
-
```

Comment:
**Thought:** Judging from [this search result](https://github.com/search?q=repo%3AGoogleCloudPlatform%2Fanthos-samples%20azurerm_subscription&type=code), there seems to be another unused instance of `data "azurerm_subscription" "current"` at: https://github.com/GoogleCloudPlatform/anthos-samples/blob/ef3baae55c6ec0e98cc062bfcfbda26345e0fe15/anthos-multi-cloud/Azure/modules/cluster-rg/main.tf#L24

Remove that is out-of-scope for this PR (but feel free to include it).

Reply:
Agreed - To keep the PR/review scope reasonable, I plan to do a similar PR for `anthos-multi-cloud`.